src/test/java/com/google/error_analyzer/backend/BooleanQueryTest.java

```diff
@@ -0,0 +1,96 @@
+/**Copyright 2019 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.*/
+
+package com.google.error_analyzer;
+
+import com.google.error_analyzer.backend.BooleanQuery;
+import com.google.error_analyzer.backend.LogDao;
+import com.google.error_analyzer.backend.LogDaoHelper;
+import com.google.error_analyzer.backend.MockLogDao;
+import com.google.error_analyzer.data.Keywords;
+import com.google.error_analyzer.data.RegexStrings;
+import java.io.IOException;
+import java.util.ArrayList;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.search.SearchHits;
+import org.junit.Assert;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class BooleanQueryTest {
+
+    private final String fileName = "file"; 
+
+    @Mock
+    LogDao database;
+
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+
+    @Test
+    public void createSearchRequestWithBoolQuery() {
+        BooleanQuery boolQuery = new BooleanQuery();
+        SearchRequest searchRequest = boolQuery.createSearchRequest(fileName);
+        String[] indices = searchRequest.indices();
+        Assert.assertEquals(indices.length, 1);
+        Assert.assertEquals(indices[0], fileName);
+    }
+
+    @Test
+    public void keyWordQueryString() {
+        String actual = Keywords.getQueryString();
```

Comment:
Please test only BooleanQuery functions here.
I think you can test keyWordQueryString & regexpQueryString by checking the values in searchRequest.